Automattic/wp-calypso
# My Home: Skipping a view tells the backend which view specifically to skip

To fix this bug p1622637733400600-slack-C01VA0R5NKY
CC @daledupreez 

#### Changes proposed in this Pull Request

When view shuffling was introduced the skipping API wasn't updated to deal with the fact that determining the "current view" is no longer deterministic. The idea of skipping the "current view" is no longer appropriate and the client will need to tell the server which view specifically the user is skipping.

* Renamed `skipCurrentViewHomeLayout` action to `skipViewHomeLayout` which now accepts an argument for which view to skip
* Add the `view` parameter to the body of the `/layout/skip` API call
* Update the Celebration, Site Setup and Task cards so they send the `view` parameter

#### Testing instructions



* Add D62241-code patch to sandbox and proxy `public-api`
* When skipping cards check that the `view` param is now being sent in the body of the `/layout/skip` API call

Dismissing one of these celebration cards with the primary CTA will skip the view
![Screen Shot on 2021-06-03 at 19:17:23](https://user-images.githubusercontent.com/1500769/120603741-6ef08500-c4a0-11eb-83a0-c37e181195c4.png)

Dismissing the very last task in the setup list will skip the `VIEW_SITE_SETUP` view
![Screen Shot on 2021-06-03 at 19:16:25](https://user-images.githubusercontent.com/1500769/120603749-7152df00-c4a0-11eb-88f8-fa27631b55d2.png)

Dismissing one of the task cards with the "hide this" drop down will skip the view
![Screen Shot on 2021-06-03 at 19:13:09](https://user-images.githubusercontent.com/1500769/120603753-72840c00-c4a0-11eb-9739-a9468a63df13.png)
